src/lib/Libutil/misc_utils.c

```diff
@@ -715,6 +715,66 @@ replace(char *str, char	*sub, char *repl, char	*retstr)
 	strncpy(retstr, rstr, i + 1);
 }
 
+
+/**
+ * @brief
+ *	Add esc as escape character if delim is appearing on str
+ *
+ * @param[in]	str		- input string
+ * @param[in]	delim  	- delimiter to be searched on str
+ * @param[in]   esc 	- escape character to be added if delim found on str
+ * @param[out]  retstr 	- delim replaced with and escape character+delim if found.
+ *
+ * @return void
+ */
+
+void
+escape_delimiter(char *str, char *delim, char esc, char *retstr)
+{
+	int	i=0, j=0;
+	int	delim_len;
+	char	rstr[MAXBUFLEN+1];
+
+	if (str == NULL)
+		return;
+
+	if (*str == '\0') {
+		retstr[0] = '\0';
+		return;
+	}
+
+	if ((delim == NULL || *delim == '\0') || esc == '\0') {
+		strncpy(retstr, str, strlen(str));
```

Comment:
how do we ensure there is no buffer overflow here? retstr may not have enough bytes..